src/data/sensor.ts

```diff
@@ -1,2 +1,6 @@
+import { UNAVAILABLE_STATES } from "./entity";
+
 export const SENSOR_DEVICE_CLASS_BATTERY = "battery";
 export const SENSOR_DEVICE_CLASS_TIMESTAMP = "timestamp";
+export const NOT_SET = "not_set";
+export const VALID_TIMESTAMP_STATES = UNAVAILABLE_STATES + [NOT_SET];
```

Comment:
"unknown" (or rather (`None`)) would be the correct state when it does not have a timestamp for this device class.
If it has a "random" string it's no longer a timestamp class.

Which integrations does that?

Reply:
The problem with "unknown" is that state is actually known, that confuses people. Other alternatives I thought about: `"none"` (as a string, should be rendered as "None", a bit confusing for developers) or `"off"`.

For example there're android next alarm sensor which uses `unavailable` when there's no alarm:
https://companion.home-assistant.io/docs/core/sensors/#next-alarm-sensor

I think that's semantically incorrect because data is available and causes unexpected side effects as I described with entity manager.

Reply:
For that example it should not be `unavailable` it should be `None`  ("unknown"), the next alarm is not known so the state is not known.

Reply:
There is currently nothing in the backend that would ever set "not_set" as a state. So it's not something that the frontend should react to.

Reply:
You're right there's nothing right now. I guess that's because frontend doesn't support it. I want to use this in [custom integration](https://github.com/leikoilja/ha-google-home) and provide people meaningful UX with nicely formatted timestamp and not set value which is not `Unknown` because `Unknown` is confusing and doesn't mean what it should mean.

Reply:
I agree that neither unknown or unavailable are correct. Changes for a device class should happen in the architecture issue though.

Reply:
Opened a discussion in the architecture repo: https://github.com/home-assistant/architecture/discussions/635